omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+            obj_paths = [geom.prim_path for geom in attached_obj.root_link.collision_meshes.values()]
+            assert len(obj_paths) <= 32, f"Expected obj_paths to be at most 32, got: {len(obj_paths)}"
+            # if len(obj_paths) > 5:
+            #     # Use visual paths instead
+            #     obj_paths = [geom.prim_path for geom in attached_obj.root_link.visual_meshes.values()]
+            self.mg.attach_objects_to_robot(
+                joint_state=cu_js_batch,
+                object_names=obj_paths,
+            )
+
+        # Run batched planning
+        if self.debug:
+            self.mg.store_debug_in_result = True
+
+        result = self.mg.plan_batch(cu_js_batch, ik_goal_batch, plan_cfg)
+
+        if self.debug:
+            from IPython import embed
+
+            embed()
+
+        # Detach attached object if it was attached
+        if attached_obj is not None:
+            self.mg.detach_object_from_robot()
+
+        if return_full_result:
+            return result
+
+        else:
+            succ = th.any(result.success).item()
+            if succ:
+                # Sample random successful path
+                successful_paths = result.get_successful_paths()
+                cmd_plan = successful_paths[np.random.choice(len(successful_paths))]
+                return self.path_to_joint_trajectory(path=cmd_plan, joint_names=sim_js_names)
+
+            else:
+                print("CuRobo plan did not converge to a solution.")
+
+    def path_to_joint_trajectory(self, path, joint_names=None):
+        """
+        Converts raw path from motion generator into joint trajectory sequence
+
+        Args:
+            path (JointState): Joint state path to convert into joint trajectory
+            joint_names (None or list): If specified, the individual joints to use when constructing the joint
+                trajectory. If None, will use all joints.
+
+        Returns:
+            torch.tensor: (N, D) joint trajectory tensor
+        """
+        cmd_plan = self.mg.get_full_js(path)
+        # get only joint names that are in both:
+        idx_list = []
+        common_js_names = []
+        jnts_to_idx = {name: i for i, name in enumerate(self.robot.joints.keys())}
+        joint_names = list(self.robot.joints.keys()) if joint_names is None else joint_names
+        for x in joint_names:
+            if x in cmd_plan.joint_names:
+                idx_list.append(jnts_to_idx[x])
+                common_js_names.append(x)
+
+        cmd_plan = cmd_plan.get_ordered_joint_state(common_js_names)
+        return cmd_plan.position
+
+    def convert_q_to_eef_traj(self, traj, return_aa=False):
+        """
+        Converts a joint trajectory @traj into an equivalent trajectory defined by end effector poses
+
+        Args:
+            traj (np.ndarray): (N, D)-shaped joint trajectory
+            return_aa (bool): Whether to return the interpolated orientations in quaternion or axis-angle representation
```

Comment:
nit: maybe say `axis-angle` instead of `aa`?

Reply:
done